### PR TITLE
Add config entry selector

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adds config entry selector:

image

Includes the possibility to filter/limit the list of config entries to a specific integration domain.

TODO:
 - [x] Backend PR: <https://github.com/home-assistant/core/pull/77108>
 - [x] Documentation PR: <https://github.com/home-assistant/home-assistant.io/pull/23836>
 - [x] Add to gallery

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/23836

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -417,6 +417,9 @@
         "add_device_id": "Choose device",
         "add_entity_id": "Choose entity"
       },
+      "config-entry-picker": {
+        "config_entry": "Config entry"
```
Comment (frenck) [inline src/translations/en.json:421]: "Config entry" is not a term we use in general towards the end-user. Should this be "Integration entry"? Or "Integration instance" or something similar?

Comment (balloob) [reply src/translations/en.json:421]: The tab in config is called Integrations. I guess we don't distinguish between Integration entries. I think just Integration is fine.

File: src/components/ha-config-entry-picker.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    this._comboBox?.focus();
+  }
+
+  protected firstUpdated() {
+    this._getConfigEntries();
+  }
+
+  private _rowRenderer: ComboBoxLitRenderer<ConfigEntryExtended> = (
+    item
+  ) => html`<mwc-list-item twoline graphic="icon">
```
Comment (zsarnett) [inline src/components/ha-config-entry-picker.ts:51]: Need to import

File: src/components/ha-config-entry-picker.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      @load=${this._onImageLoad}
+    />
+  </mwc-list-item>`;
+
+  protected render(): TemplateResult {
+    if (!this._configEntries) {
+      return html``;
+    }
+    return html`
+      <ha-combo-box
```
Comment (zsarnett) [inline src/components/ha-config-entry-picker.ts:77]: need to import the element (the comment above will help as you have to have two imports. One for the type and one for the element)

File: gallery/src/pages/components/ha-selector.ts [hunk 4]
```diff
@@ -124,6 +125,10 @@ const SCHEMAS: {
         selector: { attribute: { entity_id: "" } },
       },
       device: { name: "Device", selector: { device: {} } },
+      config_entry: {
+        name: "Config entry",
```
Comment (frenck) [inline gallery/src/pages/components/ha-selector.ts:129]:
```suggestion
        name: "Integration",
```

**Review by balloob — APPROVED** [review]

**Pull request merged**